### PR TITLE
feat: init dependencies

### LINKED ISSUE #2: Bootstrap the base for the component library
## Requirements

- vue-demi installed
- storybookjs installed
- basic structure to start working on components is initialized

### PR DESCRIPTION
**Pull request by lukicenturi**
Closes #2

### CODE REVIEW

**Comment by lukicenturi** [comment]
There is still some problem with the `vue-demi`, I tried it in the project with vue-3 installed, but it's not working.

File: .storybook/main.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import type { StorybookConfig } from '@storybook/vue-vite';
+const config: StorybookConfig = {
+  stories: ['../src/**/*.mdx', '../src/**/*.stories.@(js|jsx|ts|tsx)'],
+  addons: [
+    '@storybook/addon-links',
+    '@storybook/addon-essentials',
+    '@storybook/addon-interactions'
```
Comment (kelsos) [inline .storybook/main.ts:7]: What about the new [styling addon](https://storybook.js.org/blog/styling-addon-configure-styles-and-themes-in-storybook/) is there a problem with it?

File: README.md [hunk 2]
```diff
@@ -4,4 +4,62 @@ A vue component library and design system for rotki
...
+<script setup lang="ts">
+import { Button as RotkiButton } from '@rotki/ui-library';
+</script>
+
+<template>
+  <v-container>
+    <rotki-button label="This is button" outlined />
+  </v-container>
+</template>
+```
```
Comment (kelsos) [inline README.md:65]: Can we move this above the License? 
The license should be at the bottom of the file.

**Comment by lukicenturi** [comment]
> > > There is still some problem with the `vue-demi`, I tried it in the project with vue-3 installed, but it's not working.
> > 
> > 
> > I think I know why, vue-demi can't work with component written in SFC [vueuse/vue-demi#139](https://github.com/vueuse/vue-demi/issues/139)
> > All the examples given in the vue-demi have been in .ts file
> > New update: I'm trying this approach [vueuse/vue-demi#139 (comment)](https://github.com/vueuse/vue-demi/issues/139#issuecomment-1059793435)
> 
> Good let me know when to check again @lukicenturi thanks :)

I think the `vue-demi` works now.

**Comment by lukicenturi** [comment]
> > > > There is still some problem with the `vue-demi`, I tried it in the project with vue-3 installed, but it's not working.
> > > 
> > > 
> > > I think I know why, vue-demi can't work with component written in SFC [vueuse/vue-demi#139](https://github.com/vueuse/vue-demi/issues/139)
> > > All the examples given in the vue-demi have been in .ts file
> > > New update: I'm trying this approach [vueuse/vue-demi#139 (comment)](https://github.com/vueuse/vue-demi/issues/139#issuecomment-1059793435)
> > 
> > 
> > Good let me know when to check again @lukicenturi thanks :)
> 
> I think the `vue-demi` works now, if anyone want to check it.

File: package.json [hunk 3]
```diff
@@ -11,14 +11,69 @@
...
+    "@babel/types": "7.21.3",
     "@rotki/eslint-config": "1.0.3",
+    "@storybook/addon-essentials": "7.0.0-rc.8",
+    "@storybook/addon-interactions": "7.0.0-rc.8",
+    "@storybook/addon-styling": "0.3.2",
+    "@storybook/addon-links": "7.0.0-rc.8",
+    "@storybook/blocks": "7.0.0-rc.8",
+    "@storybook/testing-library": "0.0.14-next.1",
+    "@storybook/vue": "7.0.0-rc.8",
+    "@storybook/vue-vite": "7.0.0-rc.8",
```
Comment (lukicenturi) [inline package.json:54]: I setup storybook by their init script, but they install the rc version.
I will try to install the stable version and see if it's working.

Comment (kelsos) [reply package.json:54]: > I setup storybook by their init script, but they install the rc version. I will try to install the stable version and see if it's working.

I think 7 is still not stable, but it doesn't make sense to go with 6

File: scripts/dist-select.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+  } else {
+    switchVersion(2);
+  }
+
+  return;
+}
+
+if (!Vue || typeof Vue.version !== 'string') {
+  console.warn(
+    '[vue3-sketch-ruler] Vue is not found. Please run "npm install vue" to install.'
```
Comment (kelsos) [inline scripts/dist-select.js:28]: this will appear in the consumer?

File: package.json [hunk 5]
```diff
@@ -11,14 +11,69 @@
...
+    "@vueuse/shared": "9.13.0",
+    "vue-demi": "0.13.11"
   },
   "devDependencies": {
+    "@babel/core": "7.21.3",
+    "@babel/types": "7.21.3",
     "@rotki/eslint-config": "1.0.3",
+    "@storybook/addon-essentials": "7.0.0-rc.8",
+    "@storybook/addon-interactions": "7.0.0-rc.8",
+    "@storybook/addon-styling": "0.3.2",
```
Comment (kelsos) [inline package.json:49]: Btw checking the releases 0.3.2 seems to be the versions for storybook 6.x

# For Storybook 7
yarn add -D @storybook/addon-styling@next

Comment (kelsos) [reply package.json:49]: > Btw checking the releases 0.3.2 seems to be the versions for storybook 6.x
> 
> The article says:
> 
> ```shell
> # For Storybook 7
> yarn add -D @storybook/addon-styling@next
> ```

If you do a new install you will see that it complains about missing peer dependencies (from 6.5.x).

**Comment by kelsos** [comment]
Btw `pnpm run storybook` is the only thing I need after `pnpm install` or do I need something else too?

image

**Comment by kelsos** [comment]
> > Btw `pnpm run storybook` is the only thing I need after `pnpm install` or do I need something else too?
> > image
> 
> hmm yeah I tried only `pnpm install` and I can run the storybook. the openBlock is thing from `vue3`, but the storybook should run with `vue2.7`🧐

Can you try on a fresh checkout? too (remove `node-modules` and try to pnpm install only with the lock file)

File: scripts/dist-select.js (hunk 4 above)
Comment (kelsos) [reply scripts/dist-select.js:28]: consumer as the one that uses `yarn|npm|pnpm install @rotki/ui-library@x` if they don't see and this is only for the developer building the package then we should change this `npm` -> `pnpm`

Comment (lukicenturi) [reply scripts/dist-select.js:28]: effect of copy pastaa 😛

**Pull request merged**